algoliasearch/index.go

```diff
@@ -445,7 +445,6 @@ func (i *Index) DeleteByQuery(query string, params Params) (res BatchRes, err er
 	copy := duplicateMap(params)
 	copy["attributesToRetrieve"] = []string{"objectID"}
 	copy["hitsPerPage"] = 1000
-	copy["distinct"] = false
```

Comment:
Ok for the `s/bool/integer/`. However, we must set it to 0 anyway because we don't want the user to set it to 1 in the original `params` map. That way, we ensure to override it.